src/Catalyst.Core.Modules.Kvm/DeltaExecutor.cs

```diff
@@ -0,0 +1,411 @@
+            var sender = GetAccountAddress(entry.Base.SenderPublicKey);
+            stateUpdate.GasUsed += gasLimit;
+            Address recipient = receiver ?? Address.OfContract(sender, _stateProvider.GetNonce(sender));
+            if (txTracer.IsTracingReceipt) txTracer.MarkAsFailed(recipient, gasLimit, Bytes.Empty, "invalid");
+        }
+        
+        private void Execute(Delta delta, ITxTracer txTracer, bool readOnly, Address recipientOverride = null)
+        {
+            // int stateSnapshot = _stateDb.TakeSnapshot();
+            // int codeSnapshot = _codeDb.TakeSnapshot();
+            // if (stateSnapshot != -1 || codeSnapshot != -1)
+            // {
+            //     if(_logger.IsError) _logger.Error($"Uncommitted state ({stateSnapshot}, {codeSnapshot}) when processing from a branch root {branchStateRoot} starting with block {suggestedBlocks[0].ToString(Block.Format.Short)}");
+            // }
+            
+            // Keccak snapshotStateRoot = _stateProvider.StateRoot;
+
+            // below is the code for supporting reorganizations - to be discussed
+            // if (branchStateRoot != null && _stateProvider.StateRoot != branchStateRoot)
+            // {
+            //     /* discarding the other branch data - chain reorganization */
+            //     Metrics.Reorganizations++;
+            //     _storageProvider.Reset();
+            //     _stateProvider.Reset();
+            //     _stateProvider.StateRoot = branchStateRoot;
+            // }
+            
+            foreach (PublicEntry publicEntry in delta.PublicEntries)
+            {
+                ContractEntry contractEntry = new ContractEntry();
+                contractEntry.Base = publicEntry.Base;
+                contractEntry.Amount = publicEntry.Amount;
+                contractEntry.Data = ByteString.Empty;
+                contractEntry.GasLimit = 21000;
+                Execute(contractEntry, delta, txTracer, readOnly, recipientOverride);
+            }
+            
+            // revert state if any fails (take snapshot)
+            foreach (ContractEntry contractEntry in delta.ContractEntries)
+            {
+                Execute(contractEntry, delta, txTracer, readOnly, recipientOverride);
+            }
+
+            if (!readOnly)
+            {
+                // we should assign block rewards here (or in Ledger)
+                _stateProvider.CommitTree();
+                _storageProvider.CommitTrees();
+            }
+
+            // if ((options & ProcessingOptions.ReadOnlyChain) != 0)
+            // {
+            //     Restore(stateSnapshot, codeSnapshot, snapshotStateRoot);
+            // }
+            // else
+            // {
+            //     _stateDb.Commit();
+            //     _codeDb.Commit();
+            // }
+        }
+        
+        private void Execute(ContractEntry entry, Delta delta, ITxTracer txTracer, bool readOnly, Address recipientOverride = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The Cyclomatic Complexity of this method is 51 which is greater than 10 authorized.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4350015)